### ISSUE TITLE
forge script flag to output deployment addresses json

### ISSUE THREAD

**Issue by dcbuild3r** [foundry-rs/foundry#4499]
### Component

Forge

### Describe the feature you would like

I'd like to have something like `forge script script/deployContract.s.sol --broadcast -vvvvv --log deploy_addresses.json`. Where `deploy_addresses.json` would look like:
```js
{
contractDeploymentAddress: 0xfb3a......423c,
}
```
The motivation for this personally for me is so that I can make complex deployment scripts where I can parse the address of the deployment to start calling its functions within the script without having to manually pipe output from `forge script` to a separate file and parsing it myself.

**Comment by mds1** [comment]
Would you say this is encompassed by / a duplicate of #3911, which tracks a general `forge deploy` feature to better support complex deployment scripts?

**Comment by dcbuild3r** [comment]
yes! I need a way for forge to export the information it prints in stdout to a file. Right now I'd have to parse it manually which is prone to breaking changes etc. #3911 would definitely cover the scope of what I want and much more from what I understand by taking a quick look. Thanks @mds1
